interface/main/messages/messages.php

```diff
@@ -409,6 +426,35 @@
         $("#new_note").submit();
     }
 });
+ function gotoReport(doc_id,pname,pid,pubpid,str_dob){
+		EncounterDateArray=new Array;
+        CalendarCategoryArray=new Array;
+        EncounterIdArray=new Array;
+        Count = 0;
+	<?php
+		if(isset($enc_list) && sqlNumRows($enc_list) >0 ){
+			while($row = sqlFetchArray($enc_list)){
+	?>
+				        EncounterIdArray[Count]='<?php echo attr($row['encounter']); ?>';
+                                        EncounterDateArray[Count]='<?php echo attr(oeFormatShortDate(date("Y-m-d", strtotime($row['date'])))); ?>';
+                                        CalendarCategoryArray[Count]='<?php echo attr(xl_appt_category($row['pc_catname'])); ?>';
+                                        Count++;
+	<?php
+			}
+		}
+	?>
+	$.ajax({
+		type:'post',
+		url:'<?php echo $GLOBALS['webroot']."/library/ajax/set_session_ajax.php";?>',
+		data:{set_pid: pid},
+		async: false
+	});
+	parent.left_nav.setPatient(pname,pid,pubpid,window.name,str_dob);
+	parent.left_nav.setPatientEncounter(EncounterIdArray,EncounterDateArray,CalendarCategoryArray);
+	var baseurl  = '<?php  echo $GLOBALS['webroot'] . "/controller.php?document&view"; ?>';
+	var params   = "&patient_id=" + pid + "&document_id=" + doc_id + "&";
+	location.href = baseurl + params;
```

Comment:
This may possibly need a little work after testing (to ensure it works in both frames mode and tabs mode). Will test it after next revision.